Student of the staff deadly scaling

dmarcuse opened the pull request:
This fixes some oversights in my initial implementation of Student of the Staff automation, avoiding console errors when not in arcane cascade and correctly scaling up the extra critical damage dice with the weapon striking runes as the deadly trait is meant to do.

stwlam commented:
I think we can do this better by making strike adjustments work for damage rolls.

TikaelSol commented:
What's the status on this? The conflicts will at least need to be cleared up (just need migrations ran on it for the new predicate structure).

@stwlam how hard would it be to get this working with AdjustStrike on damage rolls? Do you mean something like this to override the damage type added by deadly?

```json
{
    "key": "AdjustStrike",
    "mode": "override",
    "property": "damageType",
    "value": "{actor|flags.pf2e.arcaneCascade.damageType}",
    "slug": "deadly-d6",
    "predicate": [
        "whatever"
    ]
}
```

dmarcuse commented:
I'm happy to update this PR to use whichever approach is best, as long as someone can point me at some documentation, an example, or some code I can use as reference

TikaelSol commented:
Yeah, just want some clarification on what shark proposed really. For now rebase and clear those conflicts.

stwlam commented:
I'll fix the bug preventing the weapon trait from having an effect.

stwlam commented:
Actually, that it changes the damage type of the deadly die is an additional complication we're not setup to handle. Go ahead and merge this.

TikaelSol commented:
Alright @dmarcuse sorry about the long delay here. Fix the conflicts and I'll merge it.

dmarcuse commented:
@TikaelSol rebased and fixed! While working on this I also noticed that the migration from `Note` to `CriticalSpecialization` rule elements seems to have accidentally applied critical specialization for all clubs rather than just staves, so I've added a commit fixing that as well.
